src/app/InteractionModelEngine.cpp

```diff
@@ -278,6 +278,37 @@ CHIP_ERROR InteractionModelEngine::ShutdownSubscription(const ScopedNodeId & aPe
     return CHIP_ERROR_KEY_NOT_FOUND;
 }
 
+bool InteractionModelEngine::IsSubjectSubscriptionActive(const FabricIndex & aFabricIndex, const NodeId & subjectID)
+{
+    for (uint16_t index = 0; index < GetNumActiveReadHandlers(); index++)
+    {
+        ReadHandler * readHandler = ActiveHandlerAt(index);
+        if (readHandler == nullptr)
+        {
+            // Should not happen
+            continue;
+        }
+
+        if (readHandler->IsType(ReadHandler::InteractionType::Subscribe))
+        {
+            Access::SubjectDescriptor subject = readHandler->GetSubjectDescriptor();
+            if (subject.fabricIndex != aFabricIndex)
+            {
+                continue;
+            }
+
+            if (subject.authMode == Access::AuthMode::kCase)
+            {
+                if (subject.cats.CheckSubjectAgainstCATs(subjectID) || subjectID == subject.subject)
+                {
+                    return readHandler->IsActiveSubscription();
```

Comment:
Why this specific boolean?  If we're in the middle of setting up a subscription with this client, wouldn't we still want to treat that as "subscription exists"?

Reply:
In the unlikely event that this scenario happens, I would side with the fact that at the moment of the check, the subscription is not set yet. For the ICD use case, according to spec we need to check if the subscription is active or not in order to send the ICD Check-In. 

If we take the ICD use case, This check will be perform right after the device transition from Idle to active, it would be pretty unlikely that a subscription was being set-up and the device decided to go into Idle right in the middle of the subscription establishment.. Actually if that happens it should be flag as a bug and fixed.

Reply:
I'm not sure I'm convinced this is the right behavior, but I agree that transitioning out of idle while in the middle of an active exchange should not happen so it does not matter too much in practice....